### ISSUE TITLE
Twice the input size with a cont layer with a stride of 2

### ISSUE THREAD

**Issue by edubois** [davisking/dlib#1192]
Hi,

If I have an input of 32, Is there a way I can get a output of 64 with the cont layer and a filter size of 3x3?

When I read the layer_abstract.hpp, I don't see a way to do this as the padding is multiplied by 2. I would like to avoid using a larger filter size such as 4 as I think this is a bit bad for the net. Or maybe I'm missing some fundamentals.

Thanks,

**Comment by davisking** [comment]
Then use a 2x2 filter with 0 padding.

**Comment by edubois** [comment]
Yes, that's what I just realized. It sounded a bit strange as I'm using 3x3 filter on other con layers, but then I realized that my max_pool was a 2x2 filter.

**Comment by edubois** [comment]
Related question: what is the pro/con of using an cont vs an upsample layer? Do you have any advise?

**Comment by davisking** [comment]
Not really.  Try both and see what works best.
